src/utils/pathCheck.ts

```diff
@@ -0,0 +1,112 @@
+import fs from 'fs';
+import path from 'path';
+
+import semver from 'semver';
+import * as childProcess from '@/utils/childProcess';
+import resources from '@/resources';
+
+const flags: Record<string, string> = {
+  helm:    'version',
+  kim:     '-v',
+  kubectl: 'version',
+};
+const regexes: Record<string, RegExp> = {
+  // helm has to match both
+  // current: version.BuildInfo{Version:"v3.5.3", ...
+  // older:   Client: &version.Version{SemVer:"v2.16.12", ...
+  helm:    /Version.*:.*?"v(.+?)"/,
+  kim:     /version v(\S+)/,
+  kubectl: /Client Version.*?GitVersion:"v(.+?)"/,
+};
+
+export default async function shadowInfo(targetDir: string, binaryName: string): Promise<Array<string>> {
+  const notes: Array<string> = [];
+  const referencePath = resources.executable(binaryName);
+
+  try {
+    await fs.promises.access(referencePath, fs.constants.R_OK | fs.constants.X_OK);
```

Comment:
https://github.com/rancher-sandbox/rancher-desktop/pull/542#discussion_r698809796 applies here too.
(All of the things that are relevant are binaries, not scripts)